src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -1767,8 +1775,17 @@ TxQ::getTxRequiredFeeAndSeq(
 
     std::uint32_t const accountSeq = sle ? (*sle)[sfSequence] : 0;
     std::uint32_t const availableSeq = nextQueuableSeqImpl(sle, lock).value();
-
-    return {mulDiv(fee, baseFee, baseLevel).second, accountSeq, availableSeq};
+    // @Ed Hennis, @nik bougalis:
+    // How do I set the correct return type for the value_or statement?
+    // How can I typecast the value_or into XRPAmount type?
+    // I tried the below static_cast but it did not compile.
+    // Below a few layers of abstraction, XRPAmount is compatible ith uint64
+    return {
+        *mulDiv(fee, baseFee, baseLevel),
+        // .value_or(static_cast<XRPAmount>(
+        // std::numeric_limits<std::uint64_t>::max())),
```

Comment:
You should be able to use `.value_or(XRPAmount(std::numeric_limits<std::int64_t>::max()` (the XRPAmount constructor takes `int64_t` and not `uint64_t`